java/ql/src/semmle/code/java/dataflow/FlowSources.qll

```diff
@@ -106,6 +108,18 @@ private class MessageBodyReaderParameterSource extends RemoteFlowSource {
   override string getSourceType() { result = "MessageBodyReader parameter" }
 }
 
+class PlayParameterSource extends RemoteFlowSource {
+  PlayParameterSource() {
+    exists(PlayActionQueryParameter p | p = this.asParameter())
+    or
+    exists(PlayHTTPRequestHeaderMethods m |
+      m.hasName("getQueryString") and m.getAParameter() = this.asParameter()
```

Comment:
There is no difference, apologies for the confusion caused by using two different terms for essentially the same thing. 

What I meant to say is that I think we should not mark the `parameter`  but instead the `return value`  as source.

Reply:
@aibaars no worries. The method takes a string as a parameter, and that string is essentially the query parameter the person is looking for. In the example below im trying to get `token`
```
getQueryString("token")
```

Reply:
That would mark the string literal `"token"` itself or any argument to `getQueryString` as remote user input. That doesn't seem right. In your example `"token"` originates from the program source code, not from an HTTP request or other remote input.

Reply:
@aibaars got it, so my idea was even though `token` is defined from the developer, but can be technically be controlled by the attacker since `getQueryString("token")` would return value from the attacker/user input and could fall into a potentially vulnerable sink. I can remove this if this is not something we want.

Reply:
Getting the return value should be the right thing to do. If I am not mistaken the following should do the trick:
```
m.hasName("getQueryString").getAReference() = this.asExpr()
```